### PR TITLE
Trial period

### PR DESCRIPTION
**Pull request by lazarmilojevic**
Added trial period functionality, as well as tests for when trials end. After a trial ends a trial user is turned into a nontrial user when they attempt to access the user page.

### CODE REVIEW

**Review by pruthvichivukula — CHANGES_REQUESTED** [review]
add a way to start the trial of a user, no GUI or calling, just a function

**Comment by lazarmilojevic** [comment]
> add a way to start the trial of a user, no GUI or calling, just a function

The trial timer starts when the user is created. User.java line 58

File: src/main/java/SpringJPA/Model/User.java [hunk 1]
```diff
@@ -51,6 +55,21 @@ public User(String user, String pass, String mail, UserType role, long apiCallCo
         this.role = role;
         this.apiCallCount = apiCallCount;
         this.apiCallLimit = apiCallLimit;
+        this.startTime = System.currentTimeMillis();
```
Comment (pruthvichivukula) [inline src/main/java/SpringJPA/Model/User.java:58]: can you make this functionality into a public function

File: src/main/java/SpringJPA/Model/User.java [hunk 2]
```diff
@@ -51,6 +55,21 @@ public User(String user, String pass, String mail, UserType role, long apiCallCo
...
+    }
```
Comment (pruthvichivukula) [inline src/main/java/SpringJPA/Model/User.java]: can you make this functionality into a public function

**Review by pruthvichivukula — CHANGES_REQUESTED** [review]

File: src/main/java/SpringJPA/Model/User.java (hunk 2 above)
Comment (lazarmilojevic) [reply src/main/java/SpringJPA/Model/User.java]: Yes, I will do that real quick

File: src/main/java/SpringJPA/Model/User.java [hunk 3]
```diff
@@ -51,6 +55,25 @@ public User(String user, String pass, String mail, UserType role, long apiCallCo
         this.role = role;
         this.apiCallCount = apiCallCount;
         this.apiCallLimit = apiCallLimit;
+        this.startTime = System.currentTimeMillis();
```
Comment (RobertDeWolf7) [inline src/main/java/SpringJPA/Model/User.java]: Change it so new users don't automatically start a trial please

File: src/test/java/SpringJPA/UserPageTest.java [hunk 4]
```diff
@@ -107,4 +110,14 @@ public void testPremiumUser() throws Exception{
...
     }
+
+    @Test
+    @WithMockUser(username="User1")
+    @DirtiesContext
+    public void testAfterTrial() throws Exception{
+        User loadedUser = userRepository.findByUsername("User1");
+        loadedUser.setStartTime(System.currentTimeMillis() - 2592000001l);
+        this.userRepository.save(loadedUser);
+        this.mockMvc.perform(get("/user")).andDo(print()).andExpect(status().isForbidden());
```
Comment (RobertDeWolf7) [inline src/test/java/SpringJPA/UserPageTest.java:121]: You'll need to add a start trial time here

File: src/main/java/SpringJPA/Model/User.java (hunk 3 above)
Comment (lazarmilojevic) [reply src/main/java/SpringJPA/Model/User.java]: Done.

File: src/test/java/SpringJPA/UserPageTest.java (hunk 4 above)
Comment (lazarmilojevic) [reply src/test/java/SpringJPA/UserPageTest.java:121]: That works as it is. setStartTime() and startTrial() both set the starttime parameter in user. So wrong place, but I did resolve the error in another location

**Review by pruthvichivukula — APPROVED** [review]
you added the function I requested

**Review by RobertDeWolf7 — APPROVED** [review]
Good work

**Pull request merged**